internal/controller/purge_controller.go

```diff
@@ -231,12 +220,22 @@ func dropFinalizers(ctx context.Context, remoteClient client.Client,
 		resource := staleResources.Items[index]
 		resource.SetFinalizers(nil)
 		if err := remoteClient.Update(ctx, &resource); err != nil {
-			return fmt.Errorf("failed to update resource: %w", err)
+			return fmt.Errorf("failed updating resource: %w", err)
 		}
 	}
 	return nil
 }
 
-func (r *PurgeReconciler) setFinalizerWarningEvent(kyma *v1beta2.Kyma, err error) {
-	r.Event(kyma, "Warning", "SettingPurgeFinalizerError", err.Error())
+func (r *PurgeReconciler) raiseSettingPurgeFinalizerFailedEvent(kyma *v1beta2.Kyma, err error) {
+	r.Event(kyma, "Warning", "SettingPurgeFinalizerFailed", err.Error())
+}
+
+func (r *PurgeReconciler) raiseRemovingPurgeFinalizerFailedEvent(kyma *v1beta2.Kyma, err error) {
+	r.Event(kyma, "Warning", "RemovingPurgeFinalizerFailed", err.Error())
+}
```

Comment:
Reviewer Note: I separated this into two events as before we didn't distinguish whether the error occurred when setting or removing the finalizer. According to API conventions events should be specific: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events